arangod/RocksDBEngine/RocksDBIndex.cpp

```diff
@@ -327,3 +332,38 @@ RocksDBKeyBounds RocksDBIndex::getBounds(Index::IndexType type, uint64_t objectI
       THROW_ARANGO_EXCEPTION(TRI_ERROR_NOT_IMPLEMENTED);
   }
 }
+
+Result RocksDBIndex::setObjectIds(std::uint64_t plannedObjectId,
+                                  std::uint64_t plannedTempObjectId) {
+  Result res;
+  auto& server = _collection.vocbase().server();
+  auto& selector = server.getFeature<EngineSelectorFeature>();
+  auto& engine = selector.engine<RocksDBEngine>();
+
+  if (plannedObjectId == _objectId && plannedTempObjectId != _tempObjectId) {
+    TRI_ASSERT(_tempObjectId == 0 || plannedTempObjectId == 0);
+    // just temp id has changed
+    std::uint64_t oldId = _tempObjectId;
+    _tempObjectId = plannedTempObjectId;
+    if (oldId != 0) {
+      try {
+        RocksDBKeyBounds bounds = getBounds(type(), oldId, unique());
+        return rocksutils::removeLargeRange(engine.db(), bounds, true, true);
+      } catch (arangodb::basics::Exception& ex) {
+        if (ex.code() != TRI_ERROR_NOT_IMPLEMENTED) {
```

Comment:
I guess we are catching only this type of error because `RocksDBIndex::getBounds` can throw it.
But it should only throw it for unknown index types, so I am wondering if we need this special handling here. Or is there a specific reason why we are ignoring this particular error. If so, a code comment would be useful.